modules/appmenu-gtk3-module.nix

```diff
@@ -0,0 +1,23 @@
+{ inputs }: { config, lib, pkgs, ... }:
+let
+  cfg = config.chaotic;
+in
+{
+  options = {
+    chaotic.appmenu-gtk3-module.enable =
+      lib.mkOption {
+        default = false;
+        description = ''
+          Sets the proper environment variable to use appmenu-gtk3-module.
+        '';
+      };
+  };
+  config = {
+    environment.systemPackages = with pkgs; [ appmenu-gtk3-module ];
+
+    # This ensures GTK applications can load appmenu-gtk-module
+    environment.profileRelativeSessionVariables = {
+      XDG_DATA_DIRS = [ "/share/gsettings-schemas/appmenu-gtk3-module-0.7.6" ];
+    };
+  };
+}
```

Comment:
```suggestion
  cfg = config.chaotic.appmenu-gtk3-module;
in
{
  options = {
    chaotic.appmenu-gtk3-module.enable =
      lib.mkOption {
        default = false;
        description = ''
          Sets the proper environment variable to use appmenu-gtk3-module.
        '';
      };
  };
  config = {
    environment.profileRelativeSessionVariables = lib.mkIf cfg.enable {
      XDG_DATA_DIRS = [ "${pkgs.appmenu-gtk3-module}/share/gsettings-schemas/appmenu-gtk3-module-0.7.6" ];
    };
  };

Reply:
The parts that I'm sure here is: `cfg = config.chaotic.appmenu-gtk3-module;` and you were missing the super-important `lib.mkIf cfg.enable`.